src/routes/auth/+server.ts

```diff
@@ -11,62 +14,56 @@ const parseError = (error: string, component: string) => {
   return `Loading ${component}: ${error}`;
 }
 
-export async function POST({ request }: { request: Request }) {
-  const { apiKey, apiSecret } = await request.json();
-
-  if (!apiKey || !apiSecret) {
-    return json({ success: false, error: 'Missing API key or API secret' });
+const awaitRequestOrFailWith = async <T>(requestPromise: Promise<T>, component?: string) => {
+  try {
+    return await requestPromise
+  } catch(error) {
+    if (component) throw new Error(parseError((error as Error).message, component))
+    else throw new Error((error as Error).message)
   }
+};
 
-  // List API key permissions
-  const [permissions, permError] = await attempt(
-    () => listApiKeyPermissions({ apiKey, apiSecret })
-  );
-  if (permError) {
-    return json({ success: false, error: parseError(permError.msg, 'Permissions') });
-  }
+const getRelayerDependingOnPermissions = async (apiCredentials: ApiCredentials,apiKeyPermissions: ApiKeysPermissions) => {
```

Comment:
This is really short and only used once, so it can be just inlined instead of having a separate function.  That way, the `awaitRequestOrFailWith(listRelayers` part will be more visible and aligns better with the other `awaitRequestOrFailWith` calls.